### DISCUSSION TITLE
keep empty directories, which will be populated when you first run the program

### DISCUSSION THREAD

**Discussion by geeohgeegeeoh** [Homebrew/discussions#2384] [Writing Formulae/Casks] [answered]
How do I stop my attempt at a .rb formula from deleting empty directories but still appear empty, while being "valid" for mapping the link farm? 

I did ```touch #{prefix}/var/db/<pkg>/.keep``` which prevented the rmdir, but the logic for mapping the dir didn't see it as populated and didn't do the symlink farm

**Comment by carlocab** [comment, marked as answer]
Try pointing the formula to `var/"db"/name` instead of `prefix/"var/db"/name` (if your formula has a `configure` script this is usually an option like `--localstatedir` or similar), and then do
```ruby
(var/"db"/name).mkpath
```
in `post_install`.

**Reply by geeohgeegeeoh** [reply]
produces an error:

```configure: error: expected an absolute directory name for --localstatedir: var```

(its the OpenBSD coders use of configure through an autogen.sh script)

**Reply by carlocab** [reply]
`var/"db"/name` is a `Pathname` that refers to an absolute path. You need to interpolate it to substitute the value.

**Reply by geeohgeegeeoh** [reply]
Many thanks. It is as you point out a two-part problem.

in the post_install, The (var/"db"/name).mkpath worked in the external /usr/local/ path. 

Once I  fixed the main body def to run ./configure with the interpolated external path, things worked fine.
